proxy.h

```diff
@@ -16,6 +16,35 @@ namespace pro {
 
 enum class constraint_level { none, nontrivial, nothrow, trivial };
 
+struct proxy_pointer_constraints {
```

Comment:
I like it - `proxy_pointer_constraints`.

Reply:
Glad you like it ^_^